### PR TITLE
[Broker] Make PersistentTopicsBase#internalExamineMessage async

### PR DESCRIPTION
**Pull request by RobertIndie**
Master Issue: #14013

### Motivation

See #14013

### Modifications

* Make PersistentTopicsBase#internalExamineMessage async

### Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

### Documentation

  
- [x] `no-need-doc`

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 1]
```diff
@@ -2540,58 +2540,81 @@ protected Response internalPeekNthMessage(String subName, int messagePosition, b
...
+                                readEntry.complete(entry);
+                            }
+
+                            @Override
+                            public void readEntryFailed(ManagedLedgerException exception, Object ctx) {
+                                readEntry.completeExceptionally(exception);
+                            }
+                        }, null);
+                        return readEntry;
+                    } catch (Exception exception) {
```
Comment (codelipenghui) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:2597]:
```suggestion
                    } catch (ManagedLedgerException exception) {
```

**Comment by RobertIndie** [comment]
These changes of codes have been moved to https://github.com/apache/pulsar/pull/16226. Close this one.

**Pull request closed without merging**